backend/services/engine_env.py

```
@@ -57,6 +57,93 @@ def _force_compile_requested() -> bool:
+    """Latch a FlashInfer apply/runtime failure for the rest of the process,
+    same contract as ``mark_compile_runtime_failure``."""
+    global _flashinfer_runtime_failure
+    try:
+        # Import/kernel errors embed absolute paths (wheels under the user's
+        # home) — redact before latching, since the reason is logged here and
+        # re-logged on every later skip.
+        from core.failure import sanitize
+
+        reason = sanitize(reason)
+    except Exception:
+        pass
```

Ruff (0.16.1)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)